backend/routes/Terraform.ts

```diff
@@ -22,47 +22,50 @@ router.use(bodyParser.json());
-            } else {
-                res.status(400).send({
-                    message: `The URL is not accessible`
-                });
+    request.get(
+        { url: zipUrl as string, encoding: null, headers: { Authorization: req.get('Authorization') } },
+        (err, zipRes, body) => {
+            if (err) {
+                logger.error(`Error while fetching zip file: ${err}`);
+                if (zipRes) {
+                    const statusCode = zipRes.statusCode >= 400 && zipRes.statusCode < 500 ? zipRes.statusCode : 400;
+                    res.status(statusCode).send({
+                        message: `The URL is not accessible - Error ${zipRes.statusCode} ${
+                            STATUS_CODES[zipRes.statusCode]
+                        }`
+                    });
+                } else {
+                    res.status(400).send({
+                        message: `The URL is not accessible`
+                    });
+                }
+                return;
             }
-            return;
-        }
 
-        decompress(body)
-            .then((files: File[]) => {
-                const modules = _(files)
-                    .filter({ type: 'directory' })
-                    .map('path')
-                    .filter(
-                        directory =>
-                            !!find(
-                                files,
+            decompress(body)
+                .then((files: File[]) => {
+                    const modules = _(files)
+                        .filter({ type: 'directory' })
+                        .map('path')
+                        .filter(directory =>
+                            files.some(
                                 file =>
                                     file.type === 'file' && file.path.match(`^${escapeRegExp(directory)}[^/]+\\.tf$`)
                             )
-                    )
-                    .map(directory => trimEnd(directory, '/'))
-                    .sort()
-                    .value();
+                        )
+                        .map(directory => trimEnd(directory, '/'))
+                        .sort()
+                        .value();
 
-                if (modules.length) res.send(modules);
-                else res.status(400).send({ message: "Couldn't find a Terraform module in the provided package" });
-            })
-            .catch((decompressErr: any) => {
-                logger.error(`Error while decompressing zip file: ${decompressErr}`);
-                res.status(400).send({ message: 'The URL does not point to a valid ZIP file' });
-            });
-    });
+                    if (modules.length) res.send(modules);
+                    else res.status(400).send({ message: "Couldn't find a Terraform module in the provided package" });
+                })
+                .catch((decompressErr: any) => {
+                    logger.error(`Error while decompressing zip file: ${decompressErr}`);
```

Comment:
Optional:

I'm not sure if with such construct it will be rendered properly (I didn't check what actually `decompress` throws). I remember seeing sometimes "[object Object]" to be displayed in such cases.

Maybe we should go with:
```suggestion
                    logger.error('Error while decompressing zip file:', decompressErr);
```
? (see https://github.com/cloudify-cosmo/cloudify-ui-common/blob/bf4f27da339fa417570926ac8021ba25e17a7105/backend/logger.js)

It's optional, just a suggestion, or maybe something to check deeper. I don't have strong opinion here.

Reply:
I guess it's always safer as you suggested, applied.